Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer+HighLowWatermarkBackPressureStrategy.swift

```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if compiler(>=5.5.2) && canImport(_Concurrency)
+/// A high-low watermarked back-pressure strategy for a ``NIOAsyncSequenceProducer``.
+///
+/// This strategy does the following:
+/// - On yield it keeps on demanding more as long as the `highWatermark` isn't reached.
+/// - On next it starts to demand once the the `lowWatermark` is reached.
```

Comment:
Can we elaborate on this a little bit: can we say something like "as long as the number of buffered elements does not exceed `highWatermark`" and so on?